pennylane/transforms/condition.py

```diff
@@ -93,8 +102,72 @@ def qnode():
             return qml.expval(qml.PauliZ(1))
     """
 
-    @wraps(then_op)
-    def wrapper(*args, **kwargs):
-        return Conditional(measurement, then_op(*args, do_queue=False, **kwargs))
+    if inspect.isclass(then_func) and issubclass(then_func, Operator):
+
+        @wraps(then_func)
+        def wrapper(*args, **kwargs):
+            ops = []
+            then_cond_op = Conditional(measurement, then_func(*args, do_queue=False, **kwargs))
+            ops.append(then_cond_op)
+
+            if else_func:
+
+                # Copy the MV such that the external state is not changed
+                inverted_m = copy(measurement)
+                else_cond_op = Conditional(~inverted_m, else_func(*args, do_queue=False, **kwargs))
+                ops.append(else_cond_op)
+
+            return ops
+
+    elif callable(then_func):
+
+        with_meas_err = (
+            "Only quantum functions that contain no measurements can be applied conditionally."

```

Comment:
@antalszava, why not?

If you remove this validation, would the following work out of the box?

```python
def circuit1(x):
    qml.RX(x, wires=0)
    return qml.expval(qml.PauliX(0))

def circuit2(x):
    qml.RY(x, wires=0)
    return qml.expval(qml.PauliY(0))

@qml.qnode(dev)
def circuit(x):
    qml.Hadamard(wires=0)
    m = qml.measure(0)
    return qml.cond(m == 1, circuit1, circuit2)(x)
```

Reply:
@josh146 yes it would work :) I just didn't see this to be compatible with our abstraction because this was terminal measurements could come in the middle of the circuit, which we do not allow at the moment. Also, using this syntax conditioning would have to affect the terminal measurement too.

Therefore I thought users could simply just define the terminal measurements in the main qfunc.